yoki-engine-docker/src/commonMain/kotlin/org/katan/yoki/engine/docker/resource/volume/VolumeResource.kt

```diff
@@ -12,82 +20,214 @@ public class VolumeResource(
 
     /**
      * Create a volume
      *
-     * @param options see VolumeCreate for options.
+     * @param config the volume schema to be used while creating.
      * @return Volume
      *
      * @see <a href="https://docs.docker.com/engine/api/v1.41/#operation/VolumeCreate">VolumeCreate</a>
      */
-    public suspend fun create(options: Map<String, Any>): Volume {
-        require(VolumeDefinition in options) { "Volume Definition is required" }
-
-        return engine.httpClient.post("/volumes/create") {
-            body = options[VolumeDefinition]!!
+    @JvmOverloads
+    public suspend fun create(config: VolumeConfig): Volume {
+        return engine.httpClient.post("$BASE_PATH/create") {
+            body = config
         }
     }
 
     /**
      * Inspect a volume
      *
-     * @param options see VolumeInspect for options.
+     * @param id the id of the inspected volume.
      * @return Volume
      *
      * @see <a href="https://docs.docker.com/engine/api/v1.41/#operation/VolumeInspect">VolumeInspect</a>
      */
-    public suspend fun inspect(options: Map<String, Any>): Volume {
-        require(VolumeName in options) { "Volume Name is required" }
-        return engine.httpClient.get("/volumes/${options[VolumeName]}")
+    @JvmOverloads
+    public suspend fun inspect(id: String): Volume {
+        return engine.httpClient.get("$BASE_PATH/${id}")
     }
 
     /**
      * Remove a volume
      *
-     * @param options see VolumeDelete for options.
+     * @param options the options to be used while removing.
      *
      * @see <a href="https://docs.docker.com/engine/api/v1.41/#operation/VolumeDelete">VolumeDelete</a>
      */
-    public suspend fun remove(options: Map<String, Any>) {
-        require(VolumeName in options) { "Volume Name is required" }
-
-        engine.httpClient.delete<Unit>("/volumes/${options[VolumeName]}") {
-            options[VolumeForceDelete]?.let {
-                parameter(VolumeForceDelete, it)
-            }
+    @JvmOverloads
+    public suspend fun remove(id: String, options: VolumeRemove? = null) {
+        engine.httpClient.delete<Unit>("$BASE_PATH/${id}") {
+            options?.let { parameter(DELETE_FORCE, it) }
         }
     }
 
     /**
      * Delete unused volumes
      *
-     * @param options see VolumePrune for options.
+     * @param options he options to be used while pruning.
      * @return VolumePruneResponse
      *
      * @see <a href="https://docs.docker.com/engine/api/v1.41/#operation/VolumePrune">VolumePrune</a>
      */
-    public suspend fun prune(options: Map<String, Any>): VolumePruneResponse {
-        return engine.httpClient.delete("/volumes/${options[VolumeName]}") {
-            options[VolumeFilters]?.let {
-                parameter(VolumeFilters, options["filter"])
-            }
+    @JvmOverloads
+    public suspend fun prune(options: VolumePrune? = null): VolumePruneResponse {
+        return engine.httpClient.delete("$BASE_PATH/prune") {
```

Comment:
This `delete` method is wrong, `delete` is used to delete volumes.
Should use `post` instead.
https://docs.docker.com/engine/api/v1.40/#operation/VolumePrune